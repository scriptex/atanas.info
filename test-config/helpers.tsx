import { act } from 'react-dom/test-utils';
import type { FC } from 'react';
import { render, RenderResult, waitFor } from '@testing-library/react';

import * as shared from '@scripts/shared';

type ComponentType = FC<Readonly<React.ComponentProps<any>>>;

export const test = async (Component: ComponentType): Promise<RenderResult> => {
	let result!: RenderResult;

	await act(async () => {
		result = await waitFor(() => render(<Component />));
	});

	return result;
};

export const snapshotTest = (Component: ComponentType, buttonSelector?: string): void => {
	const { name } = Component;

	jest.spyOn(shared, 'formatDate');

	(shared.formatDate as jest.Mock).mockImplementation(() => 'Mock date');

	afterAll(() => {
		jest.clearAllMocks();
	});

	describe(`${name} component`, () => {
		it(`Should render the ${name} component`, async () => {
			const { asFragment, container } = await test(Component);

			expect(asFragment()).toMatchSnapshot();

			if (buttonSelector) {
				await act(async () => {
					await container.querySelectorAll<HTMLElement>(buttonSelector).forEach(button => {
						if (!button.hasAttribute('download')) {
							button.click();

							expect(asFragment()).toMatchSnapshot();
						}
					});
				});
			}
		});

		it(`Should unmount the ${name} component`, async () => {
			const { unmount, asFragment } = await test(Component);

			unmount();

			expect(asFragment()).toMatchSnapshot();
		});
	});
};

export const mockAudioContext = (): jest.Mock => {
	const mockConnect = jest.fn();

	const mockgetByteFrequencyData = jest.fn();

	const mockcreateMediaElementSource = jest.fn(() => ({
		connect: mockConnect
	}));

	const mockcreateAnalyser = jest.fn(() => ({
		connect: mockConnect,
		frequencyBinCount: [0, 1, 2],
		getByteFrequencyData: mockgetByteFrequencyData
	}));

	const mockcreateOscillator = jest.fn(() => ({
		channelCount: 2
	}));

	const mockChannelSplitterConnect = jest.fn(n => n);

	const mockcreateChannelSplitter = jest.fn(() => ({
		connect: mockChannelSplitterConnect
	}));

	const mockedAudioContext = jest.fn(() => ({
		createAnalyser: mockcreateAnalyser,
		createOscillator: mockcreateOscillator,
		createChannelSplitter: mockcreateChannelSplitter,
		createMediaElementSource: mockcreateMediaElementSource
	}));

	return mockedAudioContext;
};

export const mockFetch = (value: Record<string, string | boolean>): void => {
	global.fetch = jest.fn().mockImplementation(() =>
		Promise.resolve({
			json: () => Promise.resolve(value)
		})
	);
};
