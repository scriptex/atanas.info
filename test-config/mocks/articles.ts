import type { Asset } from 'contentful';

import { Article } from '@scripts/cms';

export const articles: Article[] = [
	{
		external: true,
		externalImage: 'https://article.one/image.png',
		index: 0,
		title: 'Article 1 title',
		url: 'https://article.one/'
	},
	{
		external: false,
		image: {
			fields: {
				file: {
					url: 'https://article.two/image.png'
				}
			}
		} as Asset,
		index: 1,
		title: 'Article 2 title',
		url: 'https://article.two/'
	}
];
