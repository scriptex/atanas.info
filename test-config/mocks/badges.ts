import { Asset } from 'contentful';

import { Badge } from '@scripts/cms';

export const badges: Badge[] = [
	{
		date: '2020-12-01T00:00+03:00',
		image: {
			fields: {
				file: {
					url: 'https://www.file1.com/image.png'
				}
			}
		} as Asset,
		index: 0,
		issuer: 'Issuer 1',
		issuerUrl: 'https://www.issuer1.com',
		title: 'Badge 1',
		url: 'https://www.url1.com'
	},
	{
		date: '2020-12-02T00:00+03:00',
		image: {
			fields: {
				file: {
					url: 'https://www.file2.com/image.png'
				}
			}
		} as Asset,
		index: 1,
		issuer: 'Issuer 2',
		issuerUrl: 'https://www.issuer2.com',
		title: 'Badge 2',
		url: 'https://www.url2.com'
	}
];
