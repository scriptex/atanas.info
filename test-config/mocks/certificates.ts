import { Certificate } from '@scripts/cms';

export const certificates: Certificate[] = [
	{
		pdf: 'https://example.com/file.pdf',
		date: '2020-11-20T00:00:00',
		index: 0,
		image: 'https://example.com/image.png',
		title: 'Certificate 1'
	}
];
