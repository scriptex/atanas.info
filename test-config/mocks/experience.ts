import { Experience } from '@scripts/cms';

export const experience: Experience[] = [
	{
		index: 0,
		title: 'Experience title 1',
		project: '<p>Experience project 1</p>',
		location: 'Location 1',
		period: 'Period start - Period end',
		details: '<p>Experience details 1</p>'
	}
];
