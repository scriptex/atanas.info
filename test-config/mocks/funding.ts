import { FundingNetwork } from '@scripts/cms';

export const funding: FundingNetwork[] = [
	{
		url: 'https://example.com/1',
		name: 'Funding 1',
		index: 0,
		matrix: 'matrix(1,0,0,1,0,0)'
	},
	{
		url: 'https://example.com/2',
		name: 'Funding 2',
		index: 1,
		matrix: 'matrix(0,1,1,0,1,1)'
	}
];
