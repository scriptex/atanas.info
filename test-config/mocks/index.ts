export * from './articles';
export * from './certificates';
export * from './experience';
export * from './funding';
export * from './lastfm-insights';
export * from './occupation';
export * from './resume';
export * from './partners';
export * from './slides';
export * from './strengths';
export * from './timeline';
export * from './videos';
