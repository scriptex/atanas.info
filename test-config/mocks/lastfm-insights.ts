import { LastFMInsights } from '@insights/utils';

export const lastFmInsights: LastFMInsights = {
	error: false,
	info: {
		user: {
			age: '39',
			album_count: '1000',
			artist_count: '1234',
			bootstrap: '',
			country: 'Bulgaria',
			gender: 'male',
			image: [],
			name: 'scriptex',
			playcount: '12345',
			playlists: '10',
			realname: 'Atanas',
			registered: {
				'#text': 123456789,
				unixtime: '123456789'
			},
			subscriber: '',
			track_count: '100',
			type: 'user',
			url: 'https://atanas.info'
		}
	},
	topAlbums: [
		{
			artist: 'Test Artist',
			images: [
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/34s/2b20f07b88e2de001eeb4ea110d6a061.png',
					size: 'small'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/64s/2b20f07b88e2de001eeb4ea110d6a061.png',
					size: 'medium'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/174s/2b20f07b88e2de001eeb4ea110d6a061.png',
					size: 'large'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/300x300/2b20f07b88e2de001eeb4ea110d6a061.png',
					size: 'extralarge'
				}
			],
			name: 'Test top album'
		}
	],
	updated: 1675168164186,
	weeklyAlbumChart: [
		{
			artist: 'Test Artist',
			images: [
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/34s/ade81b221def43c48ba7b364058c13c2.png',
					size: 'small'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/64s/ade81b221def43c48ba7b364058c13c2.png',
					size: 'medium'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/174s/ade81b221def43c48ba7b364058c13c2.png',
					size: 'large'
				},
				{
					'#text': 'https://lastfm.freetls.fastly.net/i/u/300x300/ade81b221def43c48ba7b364058c13c2.png',
					size: 'extralarge'
				}
			],
			name: 'Test weekly album'
		}
	]
};
