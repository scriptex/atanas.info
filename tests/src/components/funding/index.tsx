import { act } from 'react';

import { fireEvent, waitFor } from '@testing-library/react';

import { Funding } from '@components';
import { snapshotTest, test } from '@test-config/helpers';
import { funding } from '@test-config/mocks';

snapshotTest(() => <Funding data={funding} />, undefined, 'Gunding');

it('Should test the Funding lifecycle', async () => {
	const FundingComponent = () => <Funding data={funding} />;

	const { asFragment, container } = await test(FundingComponent);

	expect(asFragment()).toMatchSnapshot();

	const testOnClick = async (selector: string) => {
		await waitFor(() => {
			fireEvent.click(container.querySelector(selector)!);
		});

		expect(asFragment()).toMatchSnapshot();
	};

	await testOnClick('.c-btn--small');

	await testOnClick('.c-funding__backdrop');

	await testOnClick('.c-btn--small');

	await testOnClick('.c-funding__trigger');

	await testOnClick('.c-btn--small');

	window.prompt = jest.fn();

	jest.useFakeTimers();

	await waitFor(() => {
		fireEvent.click(container.querySelector('.c-funding__crypto')!);
	});

	await act(async () => jest.runOnlyPendingTimers());

	expect(asFragment()).toMatchSnapshot();

	(window.prompt as jest.Mock).mockClear();
});
