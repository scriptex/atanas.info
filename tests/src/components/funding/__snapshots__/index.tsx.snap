// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gunding component Should render the Gunding component 1`] = `
<DocumentFragment>
  <div
    class="c-funding"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          +
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Gunding component Should unmount the Gunding component 1`] = `<DocumentFragment />`;

exports[`Should test the Funding lifecycle 1`] = `
<DocumentFragment>
  <div
    class="c-funding"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          +
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 2`] = `
<DocumentFragment>
  <div
    class="c-funding c-funding--open"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          -
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 3`] = `
<DocumentFragment>
  <div
    class="c-funding"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          +
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 4`] = `
<DocumentFragment>
  <div
    class="c-funding c-funding--open"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          -
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 5`] = `
<DocumentFragment>
  <div
    class="c-funding"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          +
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 6`] = `
<DocumentFragment>
  <div
    class="c-funding c-funding--open"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          -
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Should test the Funding lifecycle 7`] = `
<DocumentFragment>
  <div
    class="c-funding c-funding--open"
  >
    <button
      class="c-btn c-btn--small"
      type="button"
    >
      Sponsor me
    </button>
    <div
      class="c-funding__backdrop"
    />
    <svg
      class="c-funding__knob"
      viewBox="-100 -100 700 700"
    >
      <g>
        <g
          class="c-funding__item"
          transform="matrix(1,0,0,1,0,0)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 1
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 1"
              y="145"
            />
          </a>
        </g>
        <g
          class="c-funding__item"
          transform="matrix(0,1,1,0,1,1)"
        >
          <path
            d="M250,250 l250,0 A250,250 0 0,0 375,33.49364905389035 z"
            fill="none"
          />
          <a
            href="https://example.com/2"
            rel="noopener noreferrer"
            target="_blank"
          >
            <text
              fill="var(--color-secondary)"
              font-size="1em"
              text-anchor="middle"
              transform="rotate(60 397.2243347167969 165)"
              x="398"
              y="200"
            >
              Funding 2
            </text>
            <use
              height="40"
              transform="rotate(60 397.2243347167969 165)"
              width="40"
              x="377.2243347167969"
              xlink:href="#svg-funding 2"
              y="145"
            />
          </a>
        </g>
      </g>
      <g
        class="c-funding__trigger"
        role="button"
      >
        <circle
          cx="250"
          cy="250"
          r="30"
        />
        <text
          fill="var(--color-primary)"
          font-size="2.5em"
          text-anchor="middle"
          x="250"
          y="260"
        >
          -
        </text>
      </g>
    </svg>
    <div
      class="c-funding__crypto c-funding__crypto--bitcoin"
    >
      <h6>
        Donate Bitcoin
      </h6>
      <button
        title="Donate Bitcoin"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-bitcoin"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
    <div
      class="c-funding__crypto c-funding__crypto--shiba-inu"
    >
      <h6>
        Donate Shiba Inu
      </h6>
      <button
        title="Donate Shiba Inu"
        type="button"
      >
        <svg
          class="c-funding__crypto-icon"
        >
          <use
            xlink:href="#svg-shiba-inu"
          />
        </svg>
      </button>
      <small>
        <em>
          Click to copy wallet address
        </em>
      </small>
    </div>
  </div>
</DocumentFragment>
`;
